### PR TITLE
fix(Scripts/Quest): improve Securing the Celestial Ridge

### PR DESCRIPTION
**Pull request by grimgravy**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

_**no source needed only condition**_ 

## Issues Addressed:

- Closes 

## SOURCE: 

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. quest add 10274
2. go to the designated mission location
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Outland/zone_netherstorm.cpp [hunk 1]
```diff
@@ -949,6 +949,32 @@ class spell_q10190_battery_recharging_blaster_aura : public AuraScript
...
+{
+    NPC_VERAKU               = 18544,
+    SPELL_CHALLENGE_VERAKU   = 34895
+};
+
+class spell_challenge_veraku : public SpellScript
+{
+public:
+    PrepareSpellScript(spell_challenge_veraku);
+
```
Comment (Kitzunu) [inline src/server/scripts/Outland/zone_netherstorm.cpp:962]: Need to validate spells that are used in scripts

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: src/server/scripts/Outland/zone_netherstorm.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/scripts/Outland/zone_netherstorm.cpp:962]: SPELL_CHALLENGE_VERAKU

Comment (grimgravy) [reply src/server/scripts/Outland/zone_netherstorm.cpp:962]: I don't get it

Comment (Kitzunu) [reply src/server/scripts/Outland/zone_netherstorm.cpp:962]:
```cpp
    bool Validate(SpellInfo const* /*SpellEntry*/) override
    {
        return ValidateSpellInfo({ SPELL_CHALLENGE_VERAKU });
    }
```

Comment (grimgravy) [reply src/server/scripts/Outland/zone_netherstorm.cpp:962]: Is this normal for the condition?

Comment (Kitzunu) [reply src/server/scripts/Outland/zone_netherstorm.cpp:962]: Everytime we use a spell within a script we need to validate that it exists.

Comment (Kitzunu) [reply src/server/scripts/Outland/zone_netherstorm.cpp:962]: So this time as you use `!veraku->HasAura(SPELL_CHALLENGE_VERAKU)` you need to first validate that this spell exists before we can execute the script

Comment (Kitzunu) [reply src/server/scripts/Outland/zone_netherstorm.cpp:962]: See https://github.com/azerothcore/azerothcore-wotlk/blob/123bf37ee7e6763d9d7f4b778cfe8453577bce6c/src/server/scripts/Spells/spell_quest.cpp#L2108-L2123 for example

Comment (grimgravy) [reply src/server/scripts/Outland/zone_netherstorm.cpp:962]: understood. Thank you

**Review by Kitzunu — APPROVED** [review]

File: src/server/scripts/Outland/zone_netherstorm.cpp [hunk 2]
```diff
@@ -951,25 +951,19 @@ class spell_q10190_battery_recharging_blaster_aura : public AuraScript
...
-    {
-        return ValidateSpellInfo({ SPELL_CHALLENGE_VERAKU });
-    }
-
     SpellCastResult CheckRequirement()
     {
         if (Unit* caster = GetCaster())
             if (Creature* veraku = caster->FindNearestCreature(NPC_VERAKU, 100.0f))
-                if (!veraku->HasAura(SPELL_CHALLENGE_VERAKU))
+                if (!veraku->HasAura(GetSpellInfo()->Id))
```
Comment (Kitzunu) [inline src/server/scripts/Outland/zone_netherstorm.cpp:966]: revert this please

**Pull request merged**